src/components/SharedComponents/Sheets/TextInputSheet.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+
+import {
+  Body3, BottomSheet, Button
+} from "components/SharedComponents";
+import { View } from "components/styledComponents";
+import type { Node } from "react";
+import React, { useState } from "react";
+import { Platform } from "react-native";
+import { TextInput, useTheme } from "react-native-paper";
+import useTranslation from "sharedHooks/useTranslation";
+
+  type Props = {
+    handleClose: Function,
+    confirm: Function,
+    initialInput?: ?string,
+    placeholder: string,
+    headerText: string,
+    snapPoints: Array<Object>
+  }
+
+const TextInputSheet = ( {
+  handleClose,
+  confirm,
+  initialInput = null,
+  placeholder,
+  headerText,
+  snapPoints
+}: Props ): Node => {
+  const theme = useTheme( );
+  const [input, setInput] = useState( initialInput );
+  const { t } = useTranslation( );
+
+  return (
+    <BottomSheet
+      handleClose={handleClose}
+      headerText={headerText}
+      snapPoints={snapPoints}
+      onChange={position => {
+        if ( position === -1 ) {
+          handleClose( );
+        }
+      }}
+    >
+      <View className="p-5">
+        <TextInput
+          accessibilityLabel="Text input field"
+          keyboardType="default"
+          multiline
+          mode="flat"
+          onChangeText={text => setInput( text )}
+          value={input}
```

Comment:
Should this be `initialValue`? Not sure if the previous notes UI was working very well, but this controlled value approach is resulting in my text getting overwritten a lot.